### PR TITLE
Avoid calling torch.tensor() with tensor argument to remove warning

### LINKED ISSUE #2092: Build Warnings
#1801 seems to have introduced new build warnings which we should address.

```======================================================================================== warnings summary =========================================================================================
test/torch/tensors/test_poly.py::testSigmoid
  /Users/atrask/Laboratory/openmined/PySyft/syft/frameworks/torch/hook.py:743: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    current_tensor = hook_self.torch.native_tensor(*args, **kwargs)

test/torch/tensors/test_poly.py::testExp
  /Users/atrask/Laboratory/openmined/PySyft/syft/frameworks/torch/hook.py:743: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    current_tensor = hook_self.torch.native_tensor(*args, **kwargs)

test/torch/tensors/test_poly.py::testtanh
  /Users/atrask/Laboratory/openmined/PySyft/syft/frameworks/torch/hook.py:743: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    current_tensor = hook_self.torch.native_tensor(*args, **kwargs)

test/torch/tensors/test_poly.py::testinterpolate
  /Users/atrask/Laboratory/openmined/PySyft/syft/frameworks/torch/hook.py:743: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    current_tensor = hook_self.torch.native_tensor(*args, **kwargs)

test/torch/tensors/test_poly.py::testcustomfunction
  /Users/atrask/Laboratory/openmined/PySyft/syft/frameworks/torch/hook.py:743: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    current_tensor = hook_self.torch.native_tensor(*args, **kwargs)

test/torch/tensors/test_poly.py::testrandomfunction
  /Users/atrask/Laboratory/openmined/PySyft/syft/frameworks/torch/hook.py:743: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    current_tensor = hook_self.torch.native_tensor(*args, **kwargs)

test/torch/tensors/test_poly.py::testlogfunction
  /Users/atrask/Laboratory/openmined/PySyft/syft/frameworks/torch/hook.py:743: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    current_tensor = hook_self.torch.native_tensor(*args, **kwargs)

test/torch/tensors/test_poly.py::testexptaylor
  /Users/atrask/Laboratory/openmined/PySyft/syft/frameworks/torch/hook.py:743: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    current_tensor = hook_self.torch.native_tensor(*args, **kwargs)

test/torch/tensors/test_poly.py::testsigmoidtaylor
  /Users/atrask/Laboratory/openmined/PySyft/syft/frameworks/torch/hook.py:743: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    current_tensor = hook_self.torch.native_tensor(*args, **kwargs)

-- Docs: http://doc.pytest.org/en/latest/warnings.html```

### PR DESCRIPTION
**Pull request by midokura-silvia**
fixes #2092

### CODE REVIEW

File: syft/frameworks/torch/differential_privacy/pate.py [hunk 1]
```diff
@@ -350,7 +350,8 @@ def compute_q_noisy_max_torch(counts, noise_eps):
 
     _, winner = counts.max(0)
     counts_normalized = noise_eps * (
-        torch.tensor(counts, dtype=torch.float) - torch.tensor(counts[winner], dtype=torch.float)
+        counts.clone().detach().type(torch.float)
+        - torch.tensor(counts[winner].item(), dtype=torch.float)
```
Comment (LaRiffle) [inline syft/frameworks/torch/differential_privacy/pate.py]: Can you also `torch.tensor(counts[winner].item(), dtype=torch.float)` -> `counts[winner].clone().detach().type(torch.float)`  or it doesn't work?
Alos you might be able to call `counts = counts.float()` before to avoir type transformation

Comment (midokura-silvia) [reply syft/frameworks/torch/differential_privacy/pate.py]: counts[winner] is a single integer value. that's why calling transforming it to a float (not tensor) and then creating a tensor works. will check whether creation of tensor is necessary.
